### PR TITLE
Proxy design doc

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: design/proxy.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Proxy configuration uses three fields under the kurl section of the installer spec:
+
+```yaml
+apiVersion: cluster.kurl.sh/v1beta1
+kind: Installer
+metadata:
+  name: proxy
+spec:
+  kurl:
+    HTTPProxy: http://10.128.0.3:3128
```
Comment (marccampbell) [inline design/proxy.md:39]: How about `proxyServer` or `proxyAddress`?

File: design/proxy.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    additionalNoProxy: 10.128.04,10.128.0.5,10.128.0.6
```
Comment (marccampbell) [inline design/proxy.md:40]: Should this be array to make it easier to patch?

Comment (areed) [reply design/proxy.md:40]: @mzaneri Will array fields work with the spec and CRD?

File: design/proxy.md (hunk 1 above)
Comment (areed) [reply design/proxy.md]: changed to proxyAddress

File: design/proxy.md (hunk 2 above)
Comment (mzaneri) [reply design/proxy.md:40]: we have 2D arrays for IPTables et al cmds

Comment (mzaneri) [reply design/proxy.md:40]: I'm hesitant to change the type on this though, think its a good bit more work for not much gain

File: design/proxy.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+## Goals
+
+- Proxy installs complete successfully.
+- Docker can pull images from remote and local registries.
+
+## Non Goals
+
+- Automatically detect proxy from environment.
+- Validate kotsadm add-on works with a proxy.
+- Support docker versions below 19.03.
```
Comment (mzaneri) [inline design/proxy.md:14]: outside of scope im sure, but has anyone asked for ipv6 support for anything kURL related?

File: design/proxy.md (hunk 2 above)
Comment (mzaneri) [reply design/proxy.md:40]: Also maybe change the name to additionalProxyAddress

File: design/proxy.md (hunk 3 above)
Comment (areed) [reply design/proxy.md:14]: no, I haven't heard any requests yet

**Pull request merged**